code/src/main/java/com/onlinecontacttracing/messaging/MessageServlet.java

```diff
@@ -0,0 +1,95 @@
+
+import com.onlinecontacttracing.messaging.LocalityResource;
+import com.onlinecontacttracing.storage.CustomizableMessage;
+import com.onlinecontacttracing.storage.PositiveUser;
+import com.onlinecontacttracing.storage.PotentialContact;
+import com.onlinecontacttracing.messaging.filters.CheckMessagesForFlags;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.Projection;
+import java.util.Collections;
+import java.util.List;
+import java.util.ArrayList;
+import java.io.IOException;
+
+@WebServlet("/messaging")
+public class MessageServlet extends HttpServlet {
+  private SystemMessage systemMessage;
+  private LocalityResource localityResource;
+  private CustomizableMessage customizableMessage;
+  private String userMessage;
+  private PositiveUser user;
+  //for backend use
+  private String messagesForBackendUse;
+  //for frontend display
+  private ArrayList<String> messagesForFrontendDisplay = new ArrayList<String> ();
+  //the frontend and backend are not necessarily the same message in case the 
+  //user's message triggered flags
+
+  public MessageServlet(SystemMessage systemMessage, LocalityResource localityResource, CustomizableMessage customizableMessage, PositiveUser user) {
+    this.systemMessage = systemMessage;
+    this.localityResource = localityResource;
+    this.customizableMessage =  customizableMessage;
+    this.user = user;
+
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html;");
+    String messageLanguage = request.getParameter("language");
+    response.getWriter().println(compileMessages(messageLanguage));
+  }
+
+  public void setUserMessage() {
+    this.userMessage = this.customizableMessage.getMessage();
+  }
+
+  public void checkForFlags() {
+    CheckMessagesForFlags flagChecker = new CheckMessagesForFlags();
+    this.messagesForFrontendDisplay = CheckMessagesForFlags.findTriggeredFlags(flagChecker, this.user, this.userMessage);
+  }
+
+  //if flags are triggered, returns message and list of errors, else returns message
+  public ArrayList<String> compileMessages(String messageLanguage) {
+    String translatedResourceMessage;
+    String translatedSystemMessage;
+    this.user.incrementAttemptedEmailDrafts();
+    checkForFlags();
+    
+    if (messageLanguage.equals("SP")) {
```

Comment:
I'd prefer we represent the message language as an enum:

enum Language {
  ENGLISH, 
  SPANISH;
}